presto-orc/src/main/java/com/facebook/presto/orc/OrcReaderOptions.java

```diff
@@ -13,38 +13,35 @@
  */
 package com.facebook.presto.orc;
 
+import com.google.common.collect.ImmutableMap;
 import io.airlift.units.DataSize;
 
+import java.util.Map;
+
 import static java.util.Objects.requireNonNull;
 
 public class OrcReaderOptions
 {
     private final DataSize maxMergeDistance;
     private final DataSize tinyStripeThreshold;
     private final DataSize maxBlockSize;
-    private final boolean zstdJniDecompressionEnabled;
-    private final boolean mapNullKeysEnabled;
-    private final boolean enableTimestampMicroPrecision;
+    private final Map<OrcReaderFeature, Boolean> featureFlags;
 
     public OrcReaderOptions(DataSize maxMergeDistance, DataSize tinyStripeThreshold, DataSize maxBlockSize, boolean zstdJniDecompressionEnabled)
     {
-        this(maxMergeDistance, tinyStripeThreshold, maxBlockSize, zstdJniDecompressionEnabled, false, false);
+        this(maxMergeDistance, tinyStripeThreshold, maxBlockSize, ImmutableMap.of(OrcReaderFeature.ZSTD_JNI_DECOMPRESSION, zstdJniDecompressionEnabled));
```

Comment:
Is this used elsewhere? I slightly prefer a better encapsulated EnabledFeature class that contains isFeatureEnabled() as its method.

Reply:
You mean that simple constructor with the zstdJniDecompressionEnabled param or the param itself? If the c-tor it has 15 references in presto-orc. I opted to keep this constructor to keep PR scope smaller.